functions/go/starlark/starlark/config.go

```diff
@@ -0,0 +1,109 @@
+package starlark
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/starlark/third_party/sigs.k8s.io/kustomize/kyaml/fn/runtime/starlark"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	starlarkRunGroup      = "fn.kpt.dev"
+	starlarkRunVersion    = "v1alpha1"
+	starlarkRunAPIVersion = starlarkRunGroup + "/" + starlarkRunVersion
+	starlarkRunKind       = "StarlarkRun"
+
+	configMapApiVersion = "v1"
+	configMapKind       = "ConfigMap"
+
+	sourceKey = "source"
+
+	defaultProgramName = "stalark-function-run"
+)
+
+type StarlarkRun struct {
+	yaml.ResourceMeta `json:",inline" yaml:",inline"`
+	// Source is a required field for providing a starlark script inline.
+	Source string `json:"source" yaml:"source"`
+	// Params are the parameters in key-value pairs format.
+	Params map[string]interface{} `json:"params,omitempty" yaml:"params,omitempty"`
+}
+
+func (sr *StarlarkRun) Config(fnCfg *fn.KubeObject) error {
+	switch {
+	case fnCfg.IsGVK(configMapApiVersion, configMapKind):
+		cm := &corev1.ConfigMap{}
+		if err := fnCfg.As(cm); err != nil {
+			return err
+		}
+		// Convert ConfigMap to StarlarkRun
+		sr.Name = cm.Name
+		sr.Namespace = cm.Namespace
+		sr.Params = map[string]interface{}{}
+		for k, v := range cm.Data {
+			if k == sourceKey {
+				sr.Source = v
+			}
+			sr.Params[k] = v
+		}
+	case fnCfg.IsGVK(starlarkRunAPIVersion, starlarkRunKind):
+		if err := fnCfg.As(sr); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("`functionConfig` must be either %v or %v, but we got: %v",
+			schema.FromAPIVersionAndKind(configMapApiVersion, configMapKind).String(),
+			schema.FromAPIVersionAndKind(starlarkRunAPIVersion, starlarkRunKind).String(),
+			schema.FromAPIVersionAndKind(fnCfg.GetAPIVersion(), fnCfg.GetKind()).String())
+	}
+
+	// Defaulting
+	if sr.Name == "" {
+		sr.Name = defaultProgramName
+	}
+	// Validation
+	if sr.Source == "" {
```

Comment:
nit: suggest adding this logic inside the switch because ConfigMap and StarlarkRun have different source path (`data.source` vs `source`)